src/py/flwr/server/superlink/state/in_memory_state.py

```diff
@@ -251,6 +254,37 @@ def create_run(self, fab_id: str, fab_version: str) -> int:
         log(ERROR, "Unexpected run creation failure.")
         return 0
 
+    def store_server_public_private_key(
+        self, public_key: bytes, private_key: bytes
+    ) -> None:
+        """Store `server_public_key` and `server_private_key` in state."""
+        with self.lock:
+            if self.server_private_key is None and self.server_public_key is None:
+                self.server_private_key = private_key
+                self.server_public_key = public_key
```

Comment:
Should we raise an exception (instead of a silent failure) if this is called but credentials are already set?